File: frontend/pms_web/src/lib/repositories/suppliers.ts
import apiClient from "@/lib/api/client";

export interface Supplier {
  id: string;
  name: string;
  contact_person?: string | null;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  notes?: string | null;
  is_active: boolean;
}

export interface SupplierCreate {
  name: string;
  contact_person?: string;
  phone?: string;
  email?: string;
  address?: string;
  notes?: string;
}

export const suppliersRepo = {
  list: async (includeInactive = false): Promise<Supplier[]> => {
    const { data } = await apiClient.get("/v1/suppliers", {
      params: { include_inactive: includeInactive },
    });
    return data.items;
  },
  create: async (body: SupplierCreate): Promise<Supplier> => {
    const { data } = await apiClient.post("/v1/suppliers", body);
    return data;
  },
  update: async (id: string, body: Partial<SupplierCreate & { is_active: boolean }>) => {
    const { data } = await apiClient.patch(`/v1/suppliers/${id}`, body);
    return data as Supplier;
  },
};

File: backend/services/pms_service/app/schemas/suppliers.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class SupplierCreate(BaseModel):
    name: str
    contact_person: str | None = None
    phone: str | None = None
    email: str | None = None
    address: str | None = None
    notes: str | None = None


class SupplierUpdate(BaseModel):
    name: str | None = None
    contact_person: str | None = None
    phone: str | None = None
    email: str | None = None
    address: str | None = None
    notes: str | None = None
    is_active: bool | None = None


class SupplierOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    contact_person: str | None
    phone: str | None
    email: str | None
    address: str | None
    notes: str | None
    is_active: bool
    created_at: datetime


class SupplierList(BaseModel):
    items: list[SupplierOut]
